static/app/utils/profiling/routes.tsx

```diff
@@ -200,3 +218,60 @@ export function generateProfileFlamechartRouteWithHighlightFrame({
     },
   });
 }
+
+export function generateProfileRouteFromProfileReference({
+  orgSlug,
+  projectSlug,
+  frameName,
+  framePackage,
+  reference,
+  query,
+}: {
+  frameName: string;
+  framePackage: string | undefined;
+  orgSlug: Organization['slug'];
+  projectSlug: Project['slug'] | undefined;
```

Comment:
Should this be required? That will force the caller to pass a project slug instead of silently failing

Reply:
Yeah, lets add it. It makes the return easier to work with as we dont need the null anymore